src/osd/OSD.cc

```diff
@@ -1455,6 +1459,25 @@ void OSDService::queue_for_peering(PG *pg)
   peering_wq.queue(pg);
 }
 
+void OSDService::check_completion_item() {
+  while (1) {
+    // dout(0) << __func__ << " " << __LINE__ << " enter!! " << dendl;
+    if (!osd) assert(0);
+
+    if (comp_thread_stop) {
+      dout(0) << __func__ << " " << __LINE__ << " out " << dendl;
+      break;
+    }
+    RWLock::RLocker l(osd->pg_map_lock);
+    ceph::unordered_map<spg_t, PG*>::iterator iter = osd->pg_map.begin();
+    for (; iter != osd->pg_map.end(); ++iter) {
+      PG * pg = iter->second; 
+      if (!pg) assert(0);
+      pg->lazy_completion_func(true);
+    }
+    sleep(1);
```

Comment:
@athanatos  Actually, current completion thread use polling for checking completion item.  I think using Cond.wait()  and send event when completion item occur is right answer.